src/Arcus.EventGrid/Contracts/CloudOrEventGridEvent.cs

```
@@ -0,0 +1,127 @@
+
+            _eventGridEvent = eventGridEvent;
+        }
+
+        /// <summary>
+        /// Represent this model as a <see cref="CloudEvent"/> or <c>null</c>.
+        /// </summary>
+        public CloudEvent AsCloudEvent()
+        {
+            return _cloudEvent;
+        }
+
+        /// <summary>
+        /// Represent this model as an <see cref="EventGridEvent"/> or <c>null</c>.
+        /// </summary>
+        public EventGridEvent AsEventGridEvent()
+        {
+            return _eventGridEvent;
+        }
+
+        /// <summary>
+        /// Represent this model as a <see cref="CloudEvent"/> or <c>null</c>.
+        /// </summary>
+        public static implicit operator CloudEvent(CloudOrEventGridEvent @event)
+        {
+            return @event._cloudEvent;
+        }
+
+        /// <summary>
+        /// Represent this model as an <see cref="EventGridEvent"/> or <c>null</c>.
+        /// </summary>
+        public static implicit operator EventGridEvent(CloudOrEventGridEvent @event)
+        {
+            return @event._eventGridEvent;
+        }
+
+        /// <summary>
+        /// Represent a <see cref="CloudEvent"/> as a <see cref="CloudOrEventGridEvent"/> representation.
+        /// </summary>
+        public static implicit operator CloudOrEventGridEvent(CloudEvent cloudEvent)
+        {
+            return new CloudOrEventGridEvent(cloudEvent);
+        }
+
+        /// <summary>
+        /// Represent an <see cref="EventGridEvent"/> as a <see cref="CloudOrEventGridEvent"/> representation.
+        /// </summary>
+        public static implicit operator CloudOrEventGridEvent(EventGridEvent eventGridEvent)
+        {
+            return new CloudOrEventGridEvent(eventGridEvent);
+        }
+
+        /// <summary>
+        ///     The schema version of the data object. The publisher defines the schema version.
+        /// </summary>
+        public string DataVersion => _eventGridEvent?.DataVersion;
```

Since cloud spec doesn't support this, maybe throw an exception if `_eventGridEvent` is null?

---

Yep